src/OpenTracing/SpanOptions.php

```diff
@@ -43,7 +43,17 @@ public static function create(array $options)
 
                 case 'tags':
                     foreach ($value as $tag => $tagValue) {
-                        $spanOptions->tags[] = Tag::create($tag, $tagValue);
+                        if ($tag !== (string) $tag) {
+                            throw InvalidSpanOption::invalidTag($tag);
+                        }
+
+                        if (!is_scalar($tagValue)
+                            && !(is_object($tagValue) && method_exists($tagValue, '__toString'))
```

Comment:
I'm just not a fan of doing runtime type validations on every invocation in dynamically typed languages like JavaScript or PHP. I feel like that is the job of static type checking tools. But that's just me, I'll leave it for you 2 to decide